libraries/AP_EFI/AP_EFI_Serial_Hirth.h

```diff
@@ -0,0 +1,121 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#pragma once
+
+
+#include "AP_EFI.h"
+#include "AP_EFI_Serial_Hirth.h"
+
+
+#define HIRTH_MAX_PKT_SIZE 100
+#define HIRTH_MAX_RAW_PKT_SIZE 103
+
+#define CHECKSUM_MAX 256
+
+#define SERIAL_WAIT_DURATION 100
+#define SERIAL_WAIT_TIMEOUT 100
+
+#define ENGINE_RUNNING 4
+#define THROTTLE_POSITION_FACTOR 10
+#define CRANK_SHAFT_SENSOR_OK 0x0F
+#define INJECTION_TIME_RESOLUTION 0.8
+#define FUEL_CONSUMPTION_RESOLUTION 0.1
+#define THROTTLE_POSITION_RESOLUTION 0.1
+#define KELVIN_CONVERSION_CONSTANT 273.5
```

Comment:
All of this is implementation-detail and can be dropped in the top of the .cpp file.  Nobody including this header cares about this stuff, just the class itself.  Having it out here pollutes the namespace of everyone including the file, even transitively

Reply:
moved to .cpp file